### PR TITLE
PINET-160: set up Amazon Verified Permission

### PR DESCRIPTION
**Pull request by ghun131**
## Context

Set up Amazon Verified Permission (AVP) as the first part of story PINET-160 https://thepressingly.atlassian.net/browse/PINET-160

## Description

To avoid reimplement policy store, we've decided to go with AVP for https://docs.google.com/document/d/1d7Q8AUPzRTPKge6DMwOXWL1JphSeeT21opeAJFxQqks/edit, therefore this PR is to set up AWS verification permission

## How to test

- Firstly, add these values to `.env` file
```bash
AWS_ACCESS_KEY_ID="YOUR AWS ACCESS KEY"
AWS_SECRET_ACCESS_KEY="YOUR AWS SECRET ACCESS KEY"
```
For more information about how to acquire these values, pls visit this doc https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey
- Run this repo at your local with this command `./scripts/start.dev.sh`
- Get you policy store id from your aws console
image
and pass it to `policy_store_id` , which looks like this
image

- Go to the browser at this endpoint `http://localhost:3000/entitlement/authorization`, you should see something like the screenshot below
image

### CODE REVIEW

File: packs/publisher_portal/app/controllers/authorization_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'aws_avp'
+
+class AuthorizationController < ApplicationController
+  def index
+    client = AwsAvp.init
+    schema = client.get_schema({
+      policy_store_id: "QgU3JUDgJzFADbWzUxUFFJ"
```
Comment (tnlong311) [inline packs/publisher_portal/app/controllers/authorization_controller.rb:9]: should the `policy_store_id` value be put in env file?

File: packs/publisher_portal/app/controllers/authorization_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'aws_avp'
+
+class AuthorizationController < ApplicationController
+  def index
```
Comment (tnlong311) [inline packs/publisher_portal/app/controllers/authorization_controller.rb:6]: add docstrings comment
```ruby
# GET /entitlement/authorization
def index
end
```

File: packs/publisher_portal/app/controllers/authorization_controller.rb (hunk 1 above)
Comment (ghun131) [reply packs/publisher_portal/app/controllers/authorization_controller.rb:9]: this one will be changed later for authorization flow. `get_schema` method just to prove that the connection to `aws` is successful

File: packs/publisher_portal/app/controllers/authorization_controller.rb (hunk 2 above)
Comment (ghun131) [reply packs/publisher_portal/app/controllers/authorization_controller.rb:6]: I think this is not necessary, we should use OpenAPI for documentation later

**Review by tnlong311 — DISMISSED** [review]

File: packs/publisher_portal/config/routes/entitlement.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+scope :entitlement do
```
Comment (TronTram) [inline packs/publisher_portal/config/routes/entitlement.rb:3]: We should follow the getLago structure when defining a URL. I suggest adding
```
  namespace :api do
    namespace:v1 do
       <your code>
     end
 end
 
 ```

Comment (ghun131) [reply packs/publisher_portal/config/routes/entitlement.rb]: The namespace you suggest causes failed test cases. Please help if you have better suggestion!

**Review by TronTram — APPROVED** [review]

**Pull request merged**